ionc/ion_binary.c

```diff
@@ -91,7 +91,19 @@ int ion_binary_len_var_int_64( int64_t value )
     return len;
 }
 
-int ion_binary_len_ion_float( double value )
+int ion_binary_len_ion_float_32( float value )
+{
+    int len = 0;
+    if (value == 0 && !ion_float_is_negative_zero(value)) {
+        len = 0;
+    }
+    else {
+        len = sizeof(value); // floats are IEEE 754 32 bit binary floating point
+    }
+    return len;
+}
```

Comment:
It is regrettable what we have a bunch of copy-paste here--unfortunately, the only real alternative is a macro, but I am not sure it is worth it

Reply:
Agreed :(